src/libYARP_manager/include/yarp/manager/impl/textparser.h

```diff
@@ -38,7 +38,7 @@ class TextParser
 
         std::string ret, startKeyword, endKeyword;
         size_t s, e;
-        bool   badSymbol;
+        bool   badSymbol = false;
```

Comment:
please move the variable creation inside the following if, i.e. `bool badSymbol = ret.find("$") != std::string::npos;`